core: fix "negative" routing requirements + add requirement margin for stop on closed signal (conflict detection)

bougue-pe opened the pull request:
* Fix routing requirements cases where 'begin' was later than 'end' (stop on closed signal only).
* Also a 20 s anticipation for the start of resource requirement when stopping on closed signal.

Fix: https://github.com/OpenRailAssociation/osrd/issues/8813

bougue-pe committed:
core: add requirement margin for stop on closed signal (conflict detection)

20 s anticipation for the start of resource requirement when stopping on
closed signal.

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

Khoyo commented on core/src/main/java/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractor.kt:
```diff
@@ -40,6 +40,8 @@ import mu.KotlinLogging
 
 private val logger = KotlinLogging.logger {}
 
+const val CLOSED_SIGNAL_RESERVATION_MARGIN = 20.0
```
Could you add some documentation on that constant?

bougue-pe replied:
Done in https://github.com/OpenRailAssociation/osrd/pull/9195/commits/d4e146bff6000da1b381f62fefdb24c80a92f703

bougue-pe reviewed (commented):
~✅ just tested OK on small infra & France.
🎉 thanks for all this !~
EDIT: Wrong PR

bougue-pe commented:
Please review https://github.com/OpenRailAssociation/osrd/pull/9195/commits/d4e146bff6000da1b381f62fefdb24c80a92f703: I fixed a little bit the processing of critical distance, as it could lead to use a stop that was before the signalSight in routing requirement (too late `if`): https://github.com/OpenRailAssociation/osrd/pull/9195/commits/d4e146bff6000da1b381f62fefdb24c80a92f703#diff-8078a2f57f53541732c3008fa2e15551402baba5959c846d63fa84e9d9637e4bL395
Also checked that we are not requiring before the signalSight if stop on closed-signal is shorter than 20s and close to signal-sight position.

bougue-pe committed:
core: clarify naming in routing requirements

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
core: fix "negative" routing requirements

Begin of routing requirement was sometimes placed later than the end
requirement of a zone.
This would happen when a stop on closed signal was after the considered
route (and this route was not the first one, as there is an early return).

Too early entrySignalOffset had to be fixed too.

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe commented:
@Khoyo @eckter please re-review as there is an important fix in https://github.com/OpenRailAssociation/osrd/pull/9195/commits/8e3010d5236fa6a2f031ff0a3027447aa7635af5 (and a rename to clarify in [previous commit](https://github.com/OpenRailAssociation/osrd/pull/9195/commits/4510843e92409bf35ce7c94d46f0f351df65441c)).
This bug actually prevented to trigger the bug fixed in https://github.com/OpenRailAssociation/osrd/commit/d4e146bff6000da1b381f62fefdb24c80a92f703.

Tested locally that `assertRequirementsPeriodsConsistency` does break previous to that commit.
